src/util/system.cpp

```diff
@@ -591,7 +591,7 @@ bool ArgsManager::IsArgNegated(const std::string& strArg) const
 std::string ArgsManager::GetArg(const std::string& strArg, const std::string& strDefault) const
 {
     const util::SettingsValue value = GetSetting(strArg);
-    return value.isNull() ? strDefault : value.isFalse() ? "0" : value.isTrue() ? "1" : value.get_str();
+    return value.isNull() ? strDefault : value.isFalse() ? "0" : value.isTrue() ? "1" : value.isNum() ? value.getValStr() : value.get_str();
```

Comment:
I think we've exceeded the threshold where the ternary operator is a helpful shorthand in this case. I'd really prefer to write this out in code.

Reply:
Had the same feeling, this is what I did locally:
```cpp
if (value.isNull()) return strDefault;
if (value.isBool()) return value.isFalse() ? "0" : "1";
if (value.isNum()) return value.getValStr();
return value.get_str();
```

Reply:
In some cases the ternary operator is used in place for switch-case. If this is one such case, I think it is acceptable. Though, it might be better to put each `case` onto a new line to clarify that.

Reply:
> In some cases the ternary operator is used in place for switch-case. If this is one such case, I think it is acceptable. Though, it might be better to put each `case` onto a new line to clarify that.

Yes. I was about to update this but it got merged. I don't think it is too bad, though. The line is just meant to be read as "If isNull, return default, if isFalse, return 0, if isTrue, return 1, if isNum, return getValStr, otherwise return get_str."

Reply:
For reference my clang-format seems to understand this as well and produce:

```diff
diff --git a/src/util/system.cpp b/src/util/system.cpp
index 8e45453d31..8f37b030c1 100644
--- a/src/util/system.cpp
+++ b/src/util/system.cpp
@@ -591,7 +591,11 @@ bool ArgsManager::IsArgNegated(const std::string& strArg) const
 std::string ArgsManager::GetArg(const std::string& strArg, const std::string& strDefault) const
 {
     const util::SettingsValue value = GetSetting(strArg);
-    return value.isNull() ? strDefault : value.isFalse() ? "0" : value.isTrue() ? "1" : value.isNum() ? value.getValStr() : value.get_str();
+    return value.isNull()  ? strDefault :
+           value.isFalse() ? "0" :
+           value.isTrue()  ? "1" :
+           value.isNum()   ? value.getValStr() :
+                             value.get_str();
 }
 
 int64_t ArgsManager::GetIntArg(const std::string& strArg, int64_t nDefault) const